### PR TITLE
Added required sections in sympy.codegen module.

### PR DESCRIPTION
**Pull request by Soumi7**
# Updates sympy.codegen to follow latest docstring style guide

#### References to other Issues or PRs

#19591

#### Brief description of what is fixed or changed
- Required sections in :
  - sympy/codegen/algorithms.py
  - sympy/codegen/approximations.py
  - sympy/codegen/array_utils.py
  - sympy/codegen/ast.py
  - sympy/codegen/cfunctions.py
  - sympy/codegen/fnodes.py
  - sympy/codegen/numpy_nodes.py
  - sympy/codegen/pyutils.py
  - sympy/codegen/rewriting.py
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Soumi7** [comment]
@asmeurer This branch has a merge conflict. Do let me know if I have to remove the changes I made to fnodes.py.

**Comment by asmeurer** [comment]
It looks like the merge conflict can be fixed. If you click "resolve conflicts", you can edit the file to remove the conflicts. Someone else just modified the function definition line to include explicit keyword arguments. So the correct code should be

```
def array(symbol, dim, intent=None, *, attrs=(), value=None, type=None):
    """ Convenience function for creating a Variable instance for a Fortran array.
```

File: sympy/codegen/approximations.py [hunk 1]
```diff
@@ -12,7 +12,10 @@
 """
 
 class SumApprox(Optimization):
-    """ Approximates sum by neglecting small terms
+    """ Approximates sum by neglecting small terms.
```
Comment (asmeurer) [inline sympy/codegen/approximations.py:15]: These should be changed so the """ is on its own line.

Comment (Soumi7) [reply sympy/codegen/approximations.py]: Changed.

File: sympy/codegen/rewriting.py [hunk 2]
```diff
@@ -64,14 +64,20 @@ def __init__(self, cost_function=None, priority=1):
...
 
     Parameters
     ==========
-    query : first argument passed to replace
-    value : second argument passed to replace
+
+    query :
+        First argument passed to replace.
+    value :
+        Second argument passed to replace.
```
Comment (oscarbenjamin) [inline sympy/codegen/rewriting.py:80]: Should this not say something about the types of these arguments?

**Comment by oscarbenjamin** [comment]
I left a comment but otherwise looks good

File: sympy/codegen/rewriting.py (hunk 2 above)
Comment (Soumi7) [reply sympy/codegen/rewriting.py:80]: @aktech @asmeurer @Sc0rpi0n101 Could you look at this and suggest possible descriptions for the parameters?

Comment (aktech) [reply sympy/codegen/rewriting.py:80]: Ideally it should, but we don't have it mentioned in all of the docstrings and its not mandatory, we can improve that later as well, this is an improvement from style guide point of view anyways. I am happy to merge and improve types later.

**Pull request merged**